llvm/lib/Transforms/IPO/PassManagerBuilder.cpp

```diff
@@ -675,6 +675,9 @@ void PassManagerBuilder::populateModulePassManager(
 
   MPM.add(createAnnotation2MetadataLegacyPass());
 
+  if (SYCLOptimizationMode)
```

Comment:
... if added to BackendUtil.cpp, changes here should not be needed, right? Also, the verifier should work at all opt levels and be enabled by default, I believe.